### PR TITLE
[2.8] Fix Machine Steve Type in Provisioning Verify Test Extensions

### PR DESCRIPTION
**Pull request by caliskanugur**
## Issue: 

List call uses annotation instead of the steve type.
 
## Problem

Update List call to steve type instead the annotation
 
### Automated Testing

* Test types added/modified:
    * Validation (Go Framework)

### CODE REVIEW

**Review by floatingman — APPROVED** [review]

**Comment by caliskanugur** [comment]
Backport: https://github.com/rancher/rancher/pull/43200

File: tests/framework/extensions/provisioning/verify.go [hunk 1]
```diff
@@ -338,7 +338,7 @@ func VerifyHostnameLength(t *testing.T, client *rancher.Client, clusterObject *s
 		query2, err := url.ParseQuery(fmt.Sprintf("labelSelector=%s=%s", capi.MachineDeploymentNameLabel, md.Name))
 		require.NoError(t, err)
 
-		machineResp, err := client.Steve.SteveType(machineNameAnnotation).List(query2)
+		machineResp, err := client.Steve.SteveType("cluster.x-k8s.io.machine").List(query2)
```
Comment (igomez06) [inline tests/framework/extensions/provisioning/verify.go:341]: can we make this a const?

Comment (caliskanugur) [reply tests/framework/extensions/provisioning/verify.go]: Yes, moved the const from ssh file to verify as that's more generic, and addressed

File: tests/framework/extensions/provisioning/ssh.go [hunk 2]
```diff
@@ -22,14 +22,13 @@ import (
 )
 
 const (
-	cpuUsageVar                                            = 100 // 100 is just a placeholder until we can determine an actual number. Even with cpu usage spiking it should not go past 100% cpu usage and previous issues concerning this were hitting around 130% and above
-	checkCPU                 provisioninginput.SSHTestCase = "CheckCPU"
-	checkCPUCommand                                        = "ps -C agent -o %cpu --no-header"
-	nodeReboot               provisioninginput.SSHTestCase = "NodeReboot"
-	activeState                                            = "active"
-	runningState                                           = "running"
-	machineSteveResourceType                               = "cluster.x-k8s.io.machine"
```
Comment (igomez06) [inline tests/framework/extensions/provisioning/ssh.go:31]: ssh.go does need the machineSteveResourceType?

Comment (caliskanugur) [reply tests/framework/extensions/provisioning/ssh.go:31]: Yes it does

https://github.com/rancher/rancher/blob/f338c671ba4b81e8ab437b20d5baff478fb74df6/tests/framework/extensions/provisioning/ssh.go#L63

**Review by igomez06 — APPROVED** [review]
LGTM

**Pull request merged**